.github/workflows/pr-auto-merge.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  pull_request:
+
+jobs:
+  e2e:
+    uses: ./.github/workflows/_e2e.yaml
+
+  merge_pr_when_e2e_success:
+    name: Merge PR when E2E success
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - e2e
+    if: ${{ always() && needs.e2e.result == 'success' && contains('["dependabot","github-actions"]', github.event.pull_request.user.login) }}
+    steps:
+      - name: Merge PR
+        run: |
+          gh pr merge --merge --auto "${{github.event.pull_request.html_url}}"
+
+  notify_slack_when_e2e_failed:
+    name: Notify Slack when E2E failed
+    runs-on: ubuntu-latest
+    needs:
+      - e2e
+    if: ${{ always() && needs.e2e.result == 'failure' && contains('["dependabot","github-actions"]', github.event.pull_request.user.login) }}
+    steps:
```

Comment:
```suggestion
    needs: [e2e]
    steps:
```

Reply:
related to https://github.com/vdaas/vald-client-java/pull/180/files#r1296608869